lib/nepali_calendar/fiscal_year.rb

```diff
@@ -0,0 +1,81 @@
+class NepaliCalendar::FiscalYear
+
+    attr_accessor :start_year, :end_year
+
+    def initialize(start_year, end_year)
+        @start_year = start_year
+        @end_year = end_year
+    end
+        
+        # returns start of fiscal year date in BS
+    def beginning_of_year
+        start_date = start_year.to_s.prepend('20')
+        NepaliCalendar::Calendar.new(nil, {year: start_date, month: 4, day: 1})
+    end
+        
+        # returns end of fiscal year date in BS
+    def end_of_year
+        end_date = end_year.to_s.prepend('20')
+        NepaliCalendar::Calendar.new(nil, {year: end_date, month: 3, day: NepaliCalendar::BS[end_date.to_i][3]})
+    end
+
+
+        # returns fiscal year date in AD
+    def self.fiscal_year_for_ad_date(ad_date)
+        bs_date = NepaliCalendar::BsCalendar.ad_to_bs(ad_date.year.to_s, ad_date.month.to_s, ad_date.day.to_s)
+        if bs_date.month < 4
+          fiscal_year = ((ad_date.year - 1).to_s.slice(2,2)).to_s + ad_date.year.to_s.slice(2,2).to_s
+        else
+          fiscal_year =ad_date.year.to_s.slice(2,2).to_s + ((ad_date.year + 1).to_s.slice(2,2)).to_s
+        end
+    end
+        # returns fiscal year date in BS
+    def self.fiscal_year_for_bs_date(date) # (2079, 2, 12) ==> 7879
+      bs_date = NepaliCalendar::Calendar.new(date)
+        if bs_date.month < 4 
+          fiscal_year = (bs_date.year - 1).to_s.slice(2,2).to_s + bs_date.year.to_s.slice(2,2).to_s
+        else
+          fiscal_year = bs_date.year.to_s.slice(2,2).to_s + (bs_date.year + 1).to_s.slice(2,2).to_s
+        end
+    end
+
+    # [date] -> This is a Date object (and obviously represents AD date)
+    # Returns the fiscal year represented as a string in the form of 7778.
+    def self.fiscal_year_in_bs_for_ad_date(date)
```

Comment:
Also, this looks like a dup of `self.fiscal_year_for_ad_date(ad_date)`, unless I am missing the context.

Reply:
If it is indeed a dup, and has no use case, lets delete it.